### PR TITLE
Timezone switcher

### PR DESCRIPTION
**Pull request by jakearchibald**
Not ready to merge, but the logic is ready to review. Haven't done the styling yet.

### CODE REVIEW

File: src/_includes/layouts/main/index.njk [hunk 1]
```diff
@@ -51,8 +51,8 @@
     <!-- Analytics -->
     <script>
     (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o)
+    ;a.src=g;m=document.currentScript;m.parentNode.insertBefore(a,m)
```
Comment (jakearchibald) [inline src/_includes/layouts/main/index.njk]: I don't know why this was adding before the first script on the page, but it was messing with `document.currentScript.previous`

File: confbox.config.js [hunk 2]
```diff
@@ -31,9 +31,9 @@ module.exports = {
   twitter: '@ChromiumDev',
   /**
    * Timezone of the conference, in the form [+-]HHMM.
-   * Examples: -0800, +0100, +0530
+   * Examples: -0700, +0100, +0530
    */
-  timezone: '-0800',
+  timezone: '-0700',
```
Comment (jakearchibald) [inline confbox.config.js]: heh

**Comment by jakearchibald** [comment]
Thoughts on this: I should `visibility: hidden` the schedule if local time is set, rather than having rendering switch when the JS loads.

I should add the switch to the session page too.

File: src/_includes/timezone-toggle/index.njk [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<form style="visibility:hidden">
+  Timezone:
+  <label><input name="timezone" type="radio" checked> <span>Venue time</span></label>
+  <label><input name="timezone" type="radio"> <span>Your time</span></label>
+</form>
+<script>
+  {
+    const form = document.currentScript.previousElementSibling;
```
Comment (surma) [inline src/_includes/timezone-toggle/index.njk]: (I’m not trying to be a jerk.)

Isn’t this unnecessarily clever? Is there a downside to just using an ID and `querySelector()`?

File: src/schedule/script/index.js [hunk 4]
```diff
@@ -1,10 +1,29 @@
...
-);
+import { utcOffset as venueOffset, path } from 'confbox-config:';
+import {
+  onChange,
+  get as getTimezoneOption,
+} from '../../_includes/timezone-toggle/script/option';
+
+const localOffset = new Date().getTimezoneOffset() * 60 * 1000 * -1;
+
+console.log(localOffset);
```
Comment (surma) [inline src/schedule/script/index.js]: Remove

File: src/_includes/timezone-toggle/script/index.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  function setForm() {
+    const timezone = getOption();
+    const toCheck = timezone === 'venue' ? venue : local;
+    toCheck.checked = true;
+  }
+
+  onChange(setForm);
+  setForm();
+  form.style.visibility = 'visible';
```
Comment (surma) [inline src/_includes/timezone-toggle/script/index.js]: Should we also hide the button alltogether if the user is in the same timezone as the venue?

File: src/_includes/timezone-toggle/index.njk (hunk 3 above)
Comment (jakearchibald) [reply src/_includes/timezone-toggle/index.njk]: I was trying to support the possibility of many being on the same page. I could rely on querySelectorAll and pick the last item, would that be better?

File: src/_includes/timezone-toggle/script/index.js (hunk 5 above)
Comment (jakearchibald) [reply src/_includes/timezone-toggle/script/index.js]: Good idea

File: src/schedule/script/index.js (hunk 4 above)
Comment (jakearchibald) [reply src/schedule/script/index.js]: lol

File: src/_includes/timezone-toggle/index.njk (hunk 3 above)
Comment (jakearchibald) [reply src/_includes/timezone-toggle/index.njk]: fwiw I filed https://github.com/w3c/webcomponents/issues/809#issuecomment-532525717 because I think this stuff should be supported with web components

Comment (surma) [reply src/_includes/timezone-toggle/index.njk]: That’s a good enough reason to keep it as is imo.

File: src/_includes/timezone-toggle/script/index.js (hunk 5 above)
Comment (jakearchibald) [reply src/_includes/timezone-toggle/script/index.js]: @argyleink what do you think we should do here? If I just leave it hidden, there'll be a chunk of whitespace. I could add a little inline JS to prevent it displaying at all.

File: src/schedule/style.css [hunk 6]
```diff
@@ -230,18 +250,26 @@
   color: var(--offblack);
   line-height: 1.33;
 
+  @media (min-width: 720px) {
+    font-size: 0.7rem;
+  }
```
Comment (jakearchibald) [inline src/schedule/style.css]: @argyleink I'm increasing the font size of the schedule at mobile sizes following feedback.

File: src/sessions/html-isnt-done.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ---
 layout: layouts/session/index.njk
-title: HTML isn't done! (LAPIs)
+title: HTML isn't done!
```
Comment (jakearchibald) [inline src/sessions/html-isnt-done.md]: Ah, I was going to do another PR for this, but fuck it.

**Comment by jakearchibald** [comment]
@surma this is ready now. Going to do other style changes in another PR.

**Comment by surma** [comment]
Merge at will

File: src/_includes/timezone-toggle/script/index.js (hunk 5 above)
Comment (argyleink) [reply src/_includes/timezone-toggle/script/index.js]: great idea here, love it
let's prevent it from displaying at all if they're the same timezone 👍 it's useless at that point
